packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasEstimates.ts

```diff
@@ -36,35 +39,36 @@ async function fetcher([
 }
 
 export function useGasEstimates({
-  walletAddress,
-  sourceChainId,
-  destinationChainId,
   sourceChainErc20Address,
   destinationChainErc20Address,
-  amount,
-  sourceChainBalance
+  amount
 }: {
-  walletAddress?: string
-  sourceChainId: number
-  destinationChainId: number
   sourceChainErc20Address?: string
   destinationChainErc20Address?: string
   amount: BigNumber
-  sourceChainBalance: BigNumber | null
 }): {
   gasEstimates: GasEstimates | DepositGasEstimates | undefined
   error: any
 } {
+  const [{ sourceChain, destinationChain }] = useNetworks()
+  const {
+    app: { selectedToken: token }
+  } = useAppState()
+  const { address: walletAddress } = useAccount()
+  const balance = useBalanceOnSourceChain(token)
   const { data: signer } = useSigner()
 
+  const amountToTransfer =
+    balance !== null && amount.gte(balance) ? balance : amount
```

Comment:
This ensures that we estimate gas with a valid amount even when user's input amount is larger than their balance.

We need this change because in `useMaxAmount` now we return `undefined` as maxAmount if the gas fees are not loaded

When user clicks the max button when amount > balance, the gas estimation is not done.

So the status is loading. Both parent and child chain estimated gas are `undefined`, so we return `undefined` for `maxAmount`, hence the max amount can’t be set.